File: android/app/src/main/java/com/emops/app/domain/model/Habit.kt
package com.emops.app.domain.model

data class Habit(
    val id: String = "",
    val userId: String = "",
    val name: String = "",
    val description: String = "",
    val category: String = "",
    val frequency: String = "daily",
    val customDays: List<String>? = null,
    val targetValue: Double? = null,
    val targetUnit: String? = null,
    val reminderTime: String? = null,
    val reminderEnabled: Boolean = true,
    val streakCurrent: Int = 0,
    val streakBest: Int = 0,
    val isActive: Boolean = true,
    val sortOrder: Int = 0
)

data class HabitLog(
    val id: String = "",
    val habitId: String = "",
    val userId: String = "",
    val logDate: String = "",
    val value: Double? = null,
    val isCompleted: Boolean = false,
    val notes: String? = null
)

File: ios/EMOps/Models/Habit.swift
import Foundation

enum HabitCategory: String, Codable, CaseIterable {
    case deepWork = "deep_work"
    case reliability
    case delivery
    case security
    case aiSafety = "ai_safety"
    case leadership
    case health
    case learning
}

struct Habit: Codable, Identifiable, Equatable {
    let id: String
    let userId: String
    let name: String
    let description: String
    let category: HabitCategory
    let frequency: Frequency
    let customDays: [String]?
    let targetValue: Double
    let targetUnit: TargetUnit
    let reminderTime: String?
    let reminderEnabled: Bool
    let streakCurrent: Int
    let streakBest: Int
    let isActive: Bool
    let sortOrder: Int

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case name
        case description
        case category
        case frequency
        case customDays = "custom_days"
        case targetValue = "target_value"
        case targetUnit = "target_unit"
        case reminderTime = "reminder_time"
        case reminderEnabled = "reminder_enabled"
        case streakCurrent = "streak_current"
        case streakBest = "streak_best"
        case isActive = "is_active"
        case sortOrder = "sort_order"
    }

    enum Frequency: String, Codable, CaseIterable {
        case daily
        case weekday
        case weekly
        case custom
    }

    enum TargetUnit: String, Codable, CaseIterable {
        case hours
        case count
        case boolean
        case percentage
    }
}
